golos.charge/golos.charge.cpp

```diff
@@ -77,6 +78,10 @@ void charge::useandstore(name user, symbol_code token_code, uint8_t charge_id, i
             item.symbol_stamp = k;
             item.value = new_val.data();
         });
+
+    INLINE_ACTION_SENDER(golos::publication, calcrwrdwt) (config::publish_name,
```

Comment:
This implementation adds unnecessary dependency between charge and posting smart-contracts. But you can avoid this dependency if implement in little general form. Instead of `useandstore` you can implement method `useandnotify` which get `name user, symbol_code token_code, uint8_t charge_id, int64_t price_arg, name code, name action, uint64_t data`. This method consume charge and pass it through call action `code:action` with `data, new_val`.